packages/backend/src/services/CatalogService/CatalogService.ts

```diff
@@ -915,82 +919,123 @@ export class CatalogService<
-            };
-        }
+                let availableTimeDimensions:
+                    | ReturnType<typeof getAvailableTimeDimensionsFromTables>
+                    | undefined;
 
-        return {
-            ...metric,
-            ...(availableTimeDimensions && availableTimeDimensions.length > 0
-                ? { availableTimeDimensions }
-                : {}),
-            timeDimension,
-        };
+                // If no default time dimension is defined, we can use the available time dimensions so the user can see what time dimensions are available
+                if (!defaultTimeDimension) {
+                    availableTimeDimensions =
+                        getAvailableTimeDimensionsFromTables(tables);
+                }
+
+                let timeDimension:
+                    | MetricWithAssociatedTimeDimension['timeDimension']
+                    | undefined;
+
+                if (defaultTimeDimension) {
+                    timeDimension = {
+                        field: defaultTimeDimension.field,
+                        interval: defaultTimeDimension.interval,
+                        table: metric.table,
+                    };
+                } else if (
+                    availableTimeDimensions &&
+                    availableTimeDimensions.length > 0
+                ) {
+                    const firstAvailableTimeDimension =
+                        availableTimeDimensions[0];
+
+                    if (!firstAvailableTimeDimension.isIntervalBase) {
+                        throw new Error(
+                            'The first available time dimension is not an interval base dimension',
+                        );
+                    }
+
+                    timeDimension = {
+                        field: firstAvailableTimeDimension.name,
+                        interval:
+                            timeIntervalOverride ??
+                            DEFAULT_METRICS_EXPLORER_TIME_INTERVAL,
+                        table: firstAvailableTimeDimension.table,
+                    };
+                }
+
+                return {
+                    ...metric,
+                    ...(availableTimeDimensions &&
+                    availableTimeDimensions.length > 0
+                        ? { availableTimeDimensions }
+                        : {}),
+                    timeDimension,
+                };
+            })
+            .filter(
+                (m): m is MetricWithAssociatedTimeDimension => m !== undefined,
+            );
+
+        return mappedMetrics;
+    }
+
+    async getMetric(
+        user: SessionUser,
+        projectUuid: string,
+        tableName: string,
+        metricName: string,
+        timeIntervalOverride?: TimeFrames,
+    ) {
+        const metrics = await this.getMetrics(
+            user,
+            projectUuid,
+            [{ tableName, metricName }],
+            timeIntervalOverride,
+        );
+
+        return metrics[0];
```

Comment:
We should check if metric exists and throw error when it doesn't. Since we're using the `getMetrics` function, ts type inference doesn't show it can be `undefined`

Reply:
Good point, done here! https://github.com/lightdash/lightdash/pull/12792/commits/6c358ae17ac4daf8d9b9e4f912726806e390925b